packages/playground-examples/copy/en/4-0/New JS Features/Logical Operators and Assignment.ts

```diff
@@ -0,0 +1,53 @@
+//// { compiler: { ts: "4.0.1" } }
+
+// Logical Operators and Assignment are new features in
+// JavaScript for 2020. These are a suite of new operators
+// which edit a JavaScript object.
+
+// Their goal is to re-use the concept of mathematical 
+// operators (e.g. += -= *=) but with logic instead.
+
+interface User {
+  id?: number
+  name: string
+  location: {
+      postalCode?: string
+  }
+}
+
+function updateUser(user: User) {
+  // This code can be replaced 
+  if (!user.id) user.id = 1
+
+  // Or this code:
+  user.id = user.id || 1
+
+  // With this code:
+  user.id ||= 1
+}
+
+// The suites of operators can handle deeply nesting, which 
+// can save on quite a lot of boilerplate code too.
+
+declare const user: User
+user.location.postalCode ||= "90210"
+
+// The logical assignment operators also won't accidentally
+// trigger a setter, in comparison to the long form version:
+
+user.location.postalCode = user.location.postalCode || "90210"
+//                         ^ this could trigger a setter as a part
```

Comment:
I'm confused now.
- If `postalCode` has a setter, then won't `user.location.postalCode ||= "90210"` also trigger it? 
- If not, why not?
- the caret indicates that the setter would run on the right-hand `user.location.postalCode`, which seems like it's just reading. Is the caret in the correct place?

Or is this talking about intermediate setters, like `location`?

Reply:
I took this point from the TC39 proposal, where this was a pretty explicit 'feature' - https://github.com/tc39/proposal-logical-assignment

Though I think you're right - this isn't a great example. Maybe the pitch really is just 'terser' for most peeps